### DISCUSSION TITLE
Sending delayed messages in x-dead-letter-exchange mode with publishing confirmation enabled, delayed messages can be successfully completed. However, every time a delayed message is sent, RabbitTemplate. RabbitTemplate.ReturnsCallback are triggered. Why is this happening?

### DISCUSSION THREAD

**Discussion by nomatterfine** [rabbitmq/rabbitmq-server#7919] [Other]
### Describe the bug

Sending delayed messages in x-dead-letter-exchange mode with publishing confirmation enabled, delayed messages can be successfully completed. However, every time a delayed message is sent, RabbitTemplate.ReturnsCallback are triggered. Why is this happening?

RabbitMQ 3.8.35   Erlang 23.3.4.11

### Reproduction steps

1.
![image](https://user-images.githubusercontent.com/90117583/232652971-3330b058-213c-4531-9f41-631dc3b1e1c5.png)

2.
![image](https://user-images.githubusercontent.com/90117583/232653010-9f69d9d5-257d-4dd7-9257-e7fb338cc8ff.png)

3.
![image](https://user-images.githubusercontent.com/90117583/232653059-9475fd8f-a6b5-4e65-80ee-976b751f2841.png)

...
ReturnedMessage [message=(Body:'11' MessageProperties [headers={spring_returned_message_correlation=613060e8-1f2a-4148-96f0-f1f66e7b9340}, contentType=text/plain, contentEncoding=UTF-8, contentLength=0, receivedDeliveryMode=PERSISTENT, priority=0, receivedDelay=2000, deliveryTag=0]), replyCode=312, replyText=NO_ROUTE, exchange=delayed.exchange, routingKey=delayed.routingkey]

### Expected behavior

Did I configure it incorrectly or is it that  RabbitTemplate.ReturnsCallback are only available in x-dead-letter-exchange mode? If that's the case, I should enable publishing confirmation for delayed messages.

**Comment by michaelklishin** [comment]
RabbitMQ 3.8 is entirely [out of support](https://rabbitmq.com/versions.html).

The answer to your question is right there in the delayed message exchange README:

```
Closely related to the above, the mandatory flag is not supported by this exchange: we cannot be sure that at the future publishing point in time

 * there is at least one queue we can route to
 * the original connection is still around to send a `basic.return` to
```

**Comment by michaelklishin** [comment]
In other words, since the plugin cannot know if it can send a `basic.return` in the future, and technically it did not route anything immediately, it sends a `basic.return` immediately. That does not make much sense but neither does using mandatory publishing with this exchange type.
